contributors/design-proposals/no-new-privs.md

```diff
@@ -0,0 +1,66 @@
+
+###Support in OCI runtimes
+
+Since version 0.3.0 of the OCI runtime specification, a user can specify the `noNewPrivs` boolean flag in the configuration file.
+
+More details of OCI implementation can be checked [here](https://github.com/opencontainers/runtime-spec/pull/290).
+
+###SecurityContext in Kubernetes
+
+Kubernetes defines `SecurityContext` for `Container` and `PodSecurityContext` for `PodSpec`. `SecurityContext` objects define the related security options for Kubernetes containers, e.g. selinux options.
+
+While creating a container, kubelet parses the security context object and formats the security option strings for Docker. The security options strings will finally be inserted into `ContainerCreateConfig.HostConfig.SecurityOpt` and passed to Docker. Different Kubernetes runtimes now are using different methods to parse and format the security option strings:
+* method `#getSecurityOpts` in `docker_mager_xxxx.go` for Docker runtime
+* method `#getContainerSecurityOpts` in `docker_container.go` for CRI
+
+
+##Proposal to support "no new privileges"
+
+To support "no new privileges" options in Kubernetes, it is proposed to make the following changes:
+
+###Changes of SecurityContext objects
+
+Add a new bool type field named `noNewPrivileges` to both `SecurityContext` definition and `PodSecurityContext` definition:
+* `noNewPrivileges=true` in `PodSecurityContext` means that all the containers in the pod should be run with `no-new-privileges` enabled. This should be a pod level control of `no-new-privileges` flag.
+* `noNewPrivileges` in `SecurityContext` is a container level control of `no-new-privileges` flag, and can override the pod level `noNewPrivileges` setting.
+
+By default,  `noNewPrivileges` is `false`.
+
+The change of security context API objects requires the update of corresponding Kubernetes documents, need to submit another PR to track this.
+
+###Changes of docker runtime
+
+When parsing the new `SecurityContext` object, kubelet has to take care of `noNewPrivileges` field from security context objects. Once `noNewPrivileges` is `true`, kubelet needs to change `#getSecurityOpts` method in `docker_manager_xxx.go` to add `no-new-privileges` option to `ContainerCreateConfig.HostConfig.SecurityOpt`
+
+###Changes of CRI runtime
+
+When parsing the new `SecurityContext` object, kubelet has to take care of `noNewPrivileges` field from security context objects. Once `noNewPrivileges` is `true`, kubelet needs to change `#getContainerSecurityOpts` method in `docker_container.go` to add `no-new-privileges` option to `ContainerCreateConfig.HostConfig.SecurityOpt`
```

Comment:
nit: I prefer a higher-level description since the code (e.g., the function/file names) changes frequently.

I think you need to add a new field, e.g., `AllowPrivilegeEscalation` in the CRI `LinuxSandboxSecurityContext`. Without the API change, kubelet can't tell the runtime to do this.